solls/FileReader.cpp

```diff
@@ -0,0 +1,65 @@
+#include <solls/FileReader.h>
+#include <liblangutil/Exceptions.h>
+#include <libsolutil/Exceptions.h>
+#include <libsolutil/CommonIO.h>
+
+using solidity::frontend::ReadCallback;
+using std::string;
+
+namespace solidity {
+
+// TODO: move to <solidity/interface/ReadFile.h>?
```

Comment:
This code mostly overlaps with that in solc.

Reply:
@axic That is by intention, I literally copy'n'pasted it from there with the intend to later make use of this _new_ interface in CommandLineInterface of solc, too. I hoped for an agreement here first, maybe also as a separate PR, that this one here can be based upon.

that is, extracting the commit that isolates this functionality and makes already use of it in solc, so that the LSP can make use of the same code path here, too.